### PR TITLE
chore: improve webhook unit tests

### PR DESCRIPTION
**Pull request by mevan-karu**
Related to https://github.com/openchoreo/openchoreo/issues/2998

This pull request enhances the test suites for various webhook components by improving the robustness of environment setup and teardown and significantly expanding test coverage for webhook validation and defaulting logic. The main changes include conditional envtest initialization to support running unit tests without binaries, safer teardown procedures, and comprehensive new test cases for validating object types and edge cases.

**Test environment setup and teardown improvements:**

* Added checks in all `suite_test.go` files (`component`, `componentrelease`, `componenttype`, `clustercomponenttype`, `clustertrait`, `clusterworkflow`) to skip envtest setup if required binaries are missing, allowing unit tests to run independently. [[1]](diffhunk://#diff-e055645513ba062e7e10344035a95e2b642c78efad5ceee3cd382f6deed4cb3eR49-R54) [[2]](diffhunk://#diff-456a0064696a0eca0ca1edc7c665374f486388e69ede1c09bbbeea4771b47345R49-R54) [[3]](diffhunk://#diff-7b28b3abbd073540906e7893a05f4ae3631b65d58e09b80a9b03ffb2fda38c26R49-R54) [[4]](diffhunk://#diff-6095c007c536047e15754412bd92c203e6159aa56b442974def4d023bc587b9dR49-R54) [[5]](diffhunk://#diff-aa1c095f1a91a62b3a57da63d356e1fe7d5247f56dda98f8888707414bb56111R49-R54) [[6]](diffhunk://#diff-f949ec4b1d8a27c2bfd325d4045428991217f6af8fa6b24d9579be5c5936678aR49-R54)
* Updated `AfterSuite` hooks to safely handle cases where the test environment was not initialized, preventing unnecessary errors during teardown. [[1]](diffhunk://#diff-e055645513ba062e7e10344035a95e2b642c78efad5ceee3cd382f6deed4cb3eL109-R119) [[2]](diffhunk://#diff-456a0064696a0eca0ca1edc7c665374f486388e69ede1c09bbbeea4771b47345L109-R119) [[3]](diffhunk://#diff-7b28b3abbd073540906e7893a05f4ae3631b65d58e09b80a9b03ffb2fda38c26L109-R119) [[4]](diffhunk://#diff-6095c007c536047e15754412bd92c203e6159aa56b442974def4d023bc587b9dL109-R119) [[5]](diffhunk://#diff-aa1c095f1a91a62b3a57da63d356e1fe7d5247f56dda98f8888707414bb56111L109-R119) [[6]](diffhunk://#diff-f949ec4b1d8a27c2bfd325d4045428991217f6af8fa6b24d9579be5c5936678aL109-R119)

**Expanded webhook test coverage:**

* [`component/webhook_test.go`](diffhunk://#diff-509fd3e6d8bffb46c65269883cf8d5fec17f54e8f5cfeb3d025b2288a5d424efL25-R127): Replaced placeholder and commented-out tests with comprehensive cases for defaulter and validator logic, including checks for duplicate trait instance names and type validation errors.
* [`componentrelease/webhook_test.go`](diffhunk://#diff-1a22927582c1580f1ff7de20a76525c85dd06555526e0aaa92115835d8724f9fR12): Added tests for defaulting, update, and delete validation, as well as edge cases like invalid trait parameters and helper path adjustment functions. [[1]](diffhunk://#diff-1a22927582c1580f1ff7de20a76525c85dd06555526e0aaa92115835d8724f9fR12) [[2]](diffhunk://#diff-1a22927582c1580f1ff7de20a76525c85dd06555526e0aaa92115835d8724f9fR1105-R1203)
* [`clusterworkflow/webhook_test.go`](diffhunk://#diff-62679f3962a6a99d73af37a64de6a41f290fae85eec90990e90138b7b66ed4c0R143-R194): Added tests for type validation errors, deletion scenarios, and invalid references in `WorkflowPlaneRef`.

### CODE REVIEW

File: internal/webhook/clusterworkflow/webhook_test.go [hunk 1]
```diff
@@ -140,9 +140,58 @@ var _ = Describe("ClusterWorkflow Webhook", func() {
...
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("expected a ClusterWorkflow object for the newObj"))
+		})
+	})
+
+	Context("When deleting ClusterWorkflow", func() {
+		It("Should admit deletion of a valid ClusterWorkflow", func() {
+			_, err := validator.ValidateDelete(ctx, obj)
+			Expect(err).NotTo(HaveOccurred())
+		})
```
Comment (ChathurangaKCD) [inline internal/webhook/clusterworkflow/webhook_test.go]: We can remove `validator.ValidateDelete` method. Deletion webhooks aren't used atm & not usable with gitops.

File: internal/webhook/componentrelease/suite_test.go [hunk 2]
```diff
@@ -59,9 +65,7 @@ var _ = BeforeSuite(func() {
 		},
 	}
 
-	if getFirstFoundEnvTestBinaryDir() != "" {
-		testEnv.BinaryAssetsDirectory = getFirstFoundEnvTestBinaryDir()
-	}
+	testEnv.BinaryAssetsDirectory = getFirstFoundEnvTestBinaryDir()
```
Comment (LakshanSS) [inline internal/webhook/componentrelease/suite_test.go]: you removed the if getFirstFoundEnvTestBinaryDir() != "" guard before assigning testEnv.BinaryAssetsDirectory. But some other suite files didn't get this same simplification. They are handle in different ways. Is this intentional?

File: internal/webhook/clusterworkflow/webhook_test.go (hunk 1 above)
Comment (mevan-karu) [reply internal/webhook/clusterworkflow/webhook_test.go]: Ack. Removed.

File: internal/webhook/componentrelease/suite_test.go (hunk 2 above)
Comment (mevan-karu) [reply internal/webhook/componentrelease/suite_test.go]: They've been removed mistakenly added them back.

**Review by LakshanSS — DISMISSED** [review]

**Review by chathuranga95 — APPROVED** [review]
